migrations/portal/20221003171433-migrate_toc_pp_footer.sql

```diff
@@ -0,0 +1,90 @@
+                    true
+                  )::text::bytea, 'base64'
+                )
+              ),
+              true
+            )
+FROM (
+	-- phase3 is a table looks like
+	-- APP_A,
+	-- 		{"authgear.yaml": "abc", "templates_2f_en_2f_translation.json": "iambase64string"},
+	-- 		templates_2f_en_2f_translation.json,
+	-- 		iambase64string,
+	-- 		{"app.name": "accounts"},
+	-- 		NULL,
+	--		NULL
+	-- APP_B,
+	-- 		{"authgear.yaml": "abc", "templates_2f_zh-HK_2f_translation.json": "iambase64string"},
+	-- 		templates_2f_zh-HK_2f_translation.json,
+	-- 		iamanotherbase64string,
+	-- 		{"privacy-policy-link": "https://www.authgear.com/data-privacy", "terms-of-service-link": "https://www.authgear.com/terms"},
+	-- 		"https://www.authgear.com/data-privacy",
+	--		"https://www.authgear.com/terms"
+	SELECT *, translation_json->'privacy-policy-link' AS pp,
+		translation_json->'terms-of-service-link' AS toc FROM (
+		-- phase2 is to decode the base64 string to readable string
+		-- it is a table looks like
+		-- APP_A,
+		-- 		{"authgear.yaml": "abc", "templates_2f_en_2f_translation.json": "iambase64string"},
+		-- 		templates_2f_en_2f_translation.json,
+		-- 		iambase64string,
+		-- 		{"app.name": "accounts"}
+		-- APP_B,
+		-- 		{"authgear.yaml": "abc", "templates_2f_zh-HK_2f_translation.json": "iambase64string"},
+		-- 		templates_2f_zh-HK_2f_translation.json,
+		-- 		iamanotherbase64string,
+		-- 		{"privacy-policy-link": "https://www.authgear.com/data-privacy", "terms-of-service-link": "https://www.authgear.com/terms"}
+		SELECT *, convert_from(decode(translation_data, 'base64'), 'utf-8')::jsonb AS translation_json FROM (
+			-- phase1 is to find out the value of the translation file key
+			-- it is a table looks like
+			-- APP_A,
+			-- 		{"authgear.yaml": "abc", "templates_2f_en_2f_translation.json": "iambase64string"},
+			-- 		templates_2f_en_2f_translation.json,
+			-- 		iambase64string
+			-- APP_B,
+			-- 		{"authgear.yaml": "abc", "templates_2f_zh-HK_2f_translation.json": "iamanotherbase64string"},
+			-- 		templates_2f_zh-HK_2f_translation.json,
+			-- 		iambase64string
+			SELECT *, data->>file_name AS translation_data FROM (
+				-- phase0 is to find out the translation file key in the 'data' column
+				-- it is a table looks like
+				-- APP_A,
+				-- 		{"authgear.yaml": "abc", "templates_2f_en_2f_translation.json": "iambase64string"},
+				--		templates_2f_en_2f_translation.json
+				-- APP_B,
+				-- 		{"authgear.yaml": "abc", "templates_2f_zh-HK_2f_translation.json": "iamanotherbase64string"},
+				-- 		templates_2f_zh-HK_2f_translation.json
+				SELECT * FROM (
+					SELECT id, data, jsonb_object_keys(data) AS file_name
+					FROM _portal_config_source
+				) AS phase0
+				WHERE file_name ~ 'translation.json'
+			) AS phase1
+		) AS phase2
+	) AS phase3
+) AS phase4
+WHERE phase4.id = t1.id;
+
+-- +migrate Down
+
+-- No thing needed to be migrated down.
```

Comment:
Write a migration command, not a migration SQL.

Reply:
@louischan-oursky updated :adore2x: